explainer.md

```diff
@@ -99,68 +99,153 @@ This aim of this API is to help websites provide an optimal initial experience.
-    sessionTypes: [ "temporarypersistent-usage-record",
-                    "persistent-license" ],
-  }]);
-}).catch(_ => {
-  // Try another format/key system/robustness combination.
-});
+Specifically, `rMKSA` takes a sequence of `MediaKeySystemConfigurations`, ordered by preference. Each entry may contain a sequence of initDataTypes and sequences of audio and video contentTypes with robustness. In the dictionary above, all of these sequences are reduced to single values. 
+
+This is a fundamental difference between the APIs. MediaCapabilities aims to describe the quality (smoothness and power efficiency) of support for a single pair of audio and video streams without making a decision for the caller. Callers should still order media configurations as they do with `rMKSA`, only now they walk the list themselves, calling MediaCapabiliites once for each option. These calls will return immediately with the promises resolving asynchronously.
+
+When a key system configuration is included in the `MediaDecodingConfiguration`, `mediaCapabilities.decodingInfo()` will return a promise containing the usual three booleans (`supported`, `smooth`, and `powerEfficient`) plus a `MediaKeySystemAccess` object whenever `supported = true`. The caller may use the `MediaKeySystemAccess` as they would in traditional EME to request media keys and setup encrypted media playback. This removes the need to call `rMKSA`.
+
+ Here's a sample usage:
+
+```Javascript
+let capabilitiesPromises = []
+
+// Like rMSKA(), orderedOptions is ordered from most -> least wanted.
+for (option in orderedOptions) {
+  let promise = navigator.mediaCapabilities.decodingInfo(
+    option.decodingConfig, option.keySystemConfig);
+
+  capabilitiesPromises.push(promise)
+}
+
+(async _ => {
+  // Assume this app wants a supported && smooth config.
+  let bestConfig = null;
+  for await (const mediaCapabilityInfo of capabilitiesPromises) {
+    
+    if (!mediaCapabilityInfo.supported)
+      continue;
+
+    if (!mediaCapabilityInfo.smooth)
+      continue;
+
+    bestConfig = mediaCapabilityInfo;
+    break;
+  }
+
+  if (bestConfig) {
+    let keys = await bestConfig.keySystemAccess.createMediaKeys();
+    // NOT SHOWN: rest of EME path as-is 
+  } else {
+    console.log('No smooth configs found!');
+    
+    // NOT SHOWN: More app logic here. Maybe choose the lowest
+    // resolution and framerate available.
+  }
+})();
 ```
 
-### Permissions
 
-The EME API is designed to be able to prompt the user if needed. The reasons is that the DRM systems might require some “super cookie” to identify the device or, for example, with Firefox, the browser might need to download the components on-demand.
 
-It makes a querying API such as Media Capabilities more complex because the capabilities of the device might actually depend on the action the user takes on the requestMediaKeySystemAccess’ permission prompt. The following are proposals to solve this problem.
+### Permission prompts
```

Comment:
I would keep/add something about the Permissions API: we should extend the Permissions API to allow checking if an EME call will prompt so the website isn't surprised.

It may be good to add a note mentioning that prompts for EME are fairly rare and only two browsers prompts at the moment and one of them does a once per profile prompt.